### PR TITLE
Build from WAVs

### PR DESCRIPTION
**Pull request by ThatcherC**
I've been working on the version of the code that builds the firmware using .wav files in the build tree instead of .txt sounds files, and I think it's ready to share! I've tested this with my own voice recordings and it's pretty cool - just drop .wav files with names like `SOUND_POWER_ON.wav` or `SOUND_CONNECTING.wav` into `config/_default_cfg_src_/res/en/`, run `./build.sh` in the Docker environment, and you've got a new firmware with custom sounds!

A couple notes:
- The `sbcinfo` utility is pretty handy for inspecting the intermediate (and ephemeral) SBC audio files generated by `convert.sh`. If more work is to be done in the future on parsing out the original audio files from the firmware, or generating small or higher-quality ones from .wavs, `sbcinfo` might be useful to add to the Dockerfile
- This build system is complicated!
  - What did the trick this time was putting the sound arrays in `.cpp` files so they are built and link as `.o`s with the rest of the C/C++ files in the tree. To get this to work (along with `xxd` and `convert.sh`), a couple variable names had to be changed.
- [ ] I should update the README to describe the new way to load custom sounds into the firmware

## Issues: 
- [x] ~`make clean T=open_source` does not clean out `out/open_source/config/_default_cfg_src_/res/en/*.txt`~
  - `make allclean T=open_source` does the trick so I think is okay
- [x] ~`make` no longer runs with the `-j` parallel flag~ **Fixed!**

### CODE REVIEW

**Comment by ThatcherC** [comment]
@Ralim does this seem useful? I'm a little disappointed I haven't found a way to keep the `-j` flag in the `make` command, but even without parallel compilation it's still a pretty fast build.

**Comment by Ralim** [comment]
Apologies; thought I had commented on this.

Would be curious to figure out why we cant run with multiple threads; since that can hint to issues in the build configuration.


I prefer the concept of storing the files in .wav format for ease of use rather than the odd .txt file representation. Also it means that if in future we re-work formats used during compiling etc; we can still just have the `.wav` files around.

Overall _I_ like this; though would like to solve the make issue.

**Comment by ThatcherC** [comment]
All good! I made some progress over the weekend so now the .wav files get emitted in .cpp files that can be built and linked normally, so `make -j` works again! This is a lot cleaner than my original solution and doesn't modify the existing code too much.

File: build.sh [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/bin/bash
-make -j T=open_source DEBUG=1 >log.txt 2>&1
+make T=open_source DEBUG=1 >log.txt 2>&1
```
Comment (Ralim) [inline build.sh:2]: Can you revert the removal of the `-j`

**Comment by Ralim** [comment]
I really like this; and nice to keep the fast build working along too.

I have no qualms with this going in; I think its a notable improvement.

@WebWire-NL
You did comment notes on #14 around this; would love your thoughts on this too ❤️

File: build.sh (hunk 1 above)
Comment (ThatcherC) [reply build.sh]: Yup! Good catch - thought I'd pushed that. Build is still runs successfully with the flag re-added to `build.sh`

**Comment by WebWire-NL** [comment]
@Ralim, if it makes source code bigger then i would avoid using plain wav files. I didn't try but i can imagine wav files would be big. If they aren't then just use wave files instead of the text files :) it would be a a good addition. Otherwise it's also possible to convert them to mp3 with ffmpeg once and decompress them at build time. I still think plain wav's would make the source a lot bigger. But i didn't test (didn't build anything).

**Comment by ThatcherC** [comment]
@WebWire-NL that is a good concern! Fortunately, the WAV files tend to come out smaller than the TXT files they're replacing. As an example, check out these two files:
- [SOUND_ALEXA_START.wav](https://github.com/pine64/OpenPineBuds/blob/8a407aa29a6b636d8afe407089a80fc5a3f6e1b2/config/_default_cfg_src_/res/en/SOUND_ALEXA_START.wav) (16.6 KB)
  replaces
- [SOUND_ALEXA_START.txt](https://github.com/pine64/OpenPineBuds/blob/cede4b1c01fb40af6ca2d5bede6a15a32356c5a7/config/_default_cfg_src_/res/en/SOUND_ALEXA_START.txt) (28.1 KB)

I haven't checked all of them, but it makes a bit of sense to me that replacing 5 bytes of text (like `0x55,`) with one byte of binary in the WAV would reduce files size, even with the uncompressed WAV format.

**Comment by WebWire-NL** [comment]
I made a nice spreadsheet and compared file sizes....

https://docs.google.com/spreadsheets/d/e/2PACX-1vQkW0kRGLPBx61IBEYq3MGa0YqRxiNGpxxa0lJXDA-4xHfX7qu9BIvr1Xwj0wU1YcVVBOdXRJiQk9no/pubhtml

**Comment by WebWire-NL** [comment]
Don't get me wrong, i appreciate all the work you've put in, but i see it as just two extra lines to have an even more efficient way of compressing the files! We're talking about 1 Megabyte i know, but it's a huge difference!

**Comment by ThatcherC** [comment]
@WebWire-NL that spreadsheet is a great comparison, thanks! I've put together a couple more commits that switch everything over to MP3s - they are indeed a lot smaller. Since merging my old commits will still pull the WAV files into the git history, I made a new branch where I squashed all my work into one commit so that the (large) WAVs won't be included in the commit history. The MR for this new branch is in https://github.com/pine64/OpenPineBuds/pull/28, so I'll close this one and let's move the discussion over there.

**Comment by WebWire-NL** [comment]
@ThatcherC
@Ralim
Great improvement! Nice to see all the effort you guy's put into the project! Hope i can actually help some day again by contributing code and improving the firmware.

**Pull request closed without merging**